item_test.go

```diff
@@ -9,38 +9,45 @@ import (
 )
 
 func Test_newItem(t *testing.T) {
-	item := newItem("key", 123, time.Hour)
+	item := newItem("key", 123, time.Hour, false)
 	require.NotNil(t, item)
 	assert.Equal(t, "key", item.key)
 	assert.Equal(t, 123, item.value)
 	assert.Equal(t, time.Hour, item.ttl)
+	assert.Equal(t, false, item.enableVersionTrack)
+	assert.Equal(t, int64(-1), item.version)
 	assert.WithinDuration(t, time.Now().Add(time.Hour), item.expiresAt, time.Minute)
 }
 
 func Test_Item_update(t *testing.T) {
 	item := Item[string, string]{
-		expiresAt: time.Now().Add(-time.Hour),
-		value:     "hello",
+		expiresAt:          time.Now().Add(-time.Hour),
+		value:              "hello",
+		enableVersionTrack: true,
 	}
 
 	item.update("test", time.Hour)
 	assert.Equal(t, "test", item.value)
 	assert.Equal(t, time.Hour, item.ttl)
+	assert.Equal(t, int64(1), item.Version())
```

Comment:
Since this test block is dedicated to `Item.update()`, we shouldn't test (or rely on) other methods like `item.Version()`. Let's check `item.version`, this way the test will be truly isolated.

This applies to other cases where `item.Version()` is used as well.